project_task_material_stock/models/project_task.py

```diff
@@ -177,10 +183,29 @@ def _prepare_stock_move(self):
 
     @api.multi
     def create_stock_move(self):
-        for line in self:
-            move_id = self.env['stock.move'].create(
-                line._prepare_stock_move())
-            line.stock_move_id = move_id.id
+        pick_type = self.env.ref('stock.picking_type_out')
+
+        if self:
+            task = self[0].task_id
+            location_id = task.location_source_id.id or task.project_id.location_source_id.id or self.env.ref(
+                'stock.stock_location_stock').id
+            location_dest_id = task.location_dest_id.id or task.project_id.location_dest_id.id or self.env.ref(
+                'stock.stock_location_customers').id
+
+            picking_id = task.picking_id or self.env['stock.picking'].create({
+                'origin': "{}/{}".format(task.project_id.name, task.name),
+                'partner_id': task.partner_id.id,
+                'picking_type_id': pick_type.id,
+                'location_id': location_id,
+                'location_dest_id': location_dest_id,
+            })
+
+            for line in self:
+                if not line.stock_move_id:
```

Comment:
I think is better to create a dict and fill it with all the data and then create a picking once, adding all dict data in a line on creating the picking as is shown:
```
...
'move_lines': [(0, 0, dict_with_move_date)]
...
```

Reply:
Updated